cl/lib/context_processors.py

```diff
@@ -92,3 +95,15 @@ def inject_settings(request):
 
 def inject_random_tip(request):
     return {"TIP": random.choice(info_tips)}
+
+
+def inject_broken_email_address(request: HttpRequest) -> dict:
+    """This function injects the status of the user's email address."""
+
+    # Some requests might not contain the user attribute in that case
+    # return an AnonymousUser
+    try:
+        user = request.user
+    except AttributeError:
+        user = AnonymousUser()
+    return broken_email_address(user)
```

Comment:
I'm not sure I understand the need to use AnonymousUser here at all.  If the user is anonymous, we won't have a broken email for them right?

Reply:
yes, as we talked, there was no need to use `AnonymousUser` here, since it seems that every request handled by Django has a `User` or `AnonymousUser` injected by the [auth context processor](https://github.com/django/django/blob/main/django/contrib/auth/context_processors.py#L49)
So to solve the failing test I added an AnonymousUser to the emulated HttpRequest in that [test](https://github.com/freelawproject/courtlistener/blob/94-show-broken-email-address-banner/cl/search/tests.py#L1170)